tools/Microsoft.TemplateEngine.Authoring.TemplateVerifier/Commands/IInstantiationResult.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.TemplateEngine.Authoring.TemplateVerifier.Commands;
+
+/// <summary>
+/// The result information of a template instantiation action.
+/// </summary>
+public interface IInstantiationResult
+{
+    /// <summary>
+    /// Exit code of the action (e.g. exit code of the dotnet new command).
+    /// 0 indicates successful action. Nonzero otherwise.
+    /// </summary>
+    int ExitCode { get; }
+
+    /// <summary>
+    /// Standard output stream content for the instantiation action.
+    /// </summary>
+    string StdOut { get; }
+
+    /// <summary>
+    /// Standard error stream content for the instantiation action.
+    /// </summary>
+    string StdErr { get; }
```

Comment:
just a thought - does it make sense to be in an a separate interface?
`IInstantiationResult`
`ICLIInstantiationResult` : `IInstantiationResult`